worlds/minecraft/ItemPool.py

```diff
@@ -0,0 +1,42 @@
+from math import ceil
+from . import Constants
+
+def build_item_pool(mc_world):
+	multiworld = mc_world.multiworld
+	player = mc_world.player
+
+	itempool = []
+	total_location_count = len(multiworld.get_unfilled_locations(player))
+
+	required_pool = Constants.item_info["required_pool"]
+	junk_weights = Constants.item_info["junk_weights"]
+
+	# Add required progression items
+	for item_name, num in required_pool.items():
+		itempool += [mc_world.create_item(item_name) for _ in range(num)]
+
+	# Add structure compasses
+	if multiworld.structure_compasses[player]:
+		compasses = [name for name in mc_world.item_name_to_id if "Structure Compass" in name]
+		for item_name in compasses:
+			itempool.append(mc_world.create_item(item_name))
+
+	# Dragon egg shards
+	if multiworld.egg_shards_required[player] > 0:
+		num = multiworld.egg_shards_available[player]

```

Comment:
I could be mistaken but I don't see any min/max around either of these numbers anywhere to make sure available >= required.

Reply:
Logic for this is in set_rules where it does check min of required/available for logical completion

Reply:
set_rules happens after generate_basic which is after create_items, so that's too late.